include/kernel.h

```diff
@@ -280,6 +280,22 @@ __syscall k_tid_t k_thread_create(struct k_thread *new_thread,
 				  void *p1, void *p2, void *p3,
 				  int prio, uint32_t options, k_timeout_t delay);
 
+/**
+ * @brief Dynamically allocate a thread stack from the heap.
+ *
+ * The pointer stored in @p stack will suitable to pass to
+ * @ref k_thread_create.
```

Comment:
Pedantry: you should specify that this is allocated from the general kernel heap used by k_malloc() et. al.  Zephyr has a bunch of subsystems that do their own memory management, so IMHO we should be clear about what we mean by "dynamically allocate".